tests/test_realisation.py

```diff
@@ -0,0 +1,403 @@
+                "duration": 60.0,
+                "dt": 0.005,
+            }
+        }
+    domain_parameters_read = realisations.DomainParameters.read_from_realisation(
+        realisation_ffp
+    )
+    assert domain_parameters_read.depth == domain_parameters.depth
+    assert domain_parameters_read.duration == domain_parameters.duration
+    assert domain_parameters_read.dt == domain_parameters.dt
+    assert domain_parameters_read.resolution == domain_parameters.resolution
+    assert (
+        domain_parameters_read.domain.corners == domain_parameters.domain.corners
+    ).all()
+
+
+def test_domain_parameters_properties():
+    domain_parameters = realisations.DomainParameters(
+        resolution=0.1,  # a 0.1km resolution
+        domain=bounding_box.BoundingBox.from_centroid_bearing_extents(
+            centroid=np.array([-43.53092, 172.63701]),
+            bearing=45.0,
+            extent_x=100.0,
+            extent_y=100.0,
+        ),
+        depth=40.0,
+        duration=60.0,
+        dt=0.005,
+    )
+    assert domain_parameters.nx == 1000
+    assert domain_parameters.ny == 1000
+    assert domain_parameters.nz == 400
+
+
+def test_srf_config_example(tmp_path):
+    domain_parameters = realisations.DomainParameters(
+        resolution=0.1,  # a 0.1km resolution
+        domain=bounding_box.BoundingBox.from_centroid_bearing_extents(
+            centroid=np.array([-43.53092, 172.63701]),
+            bearing=45.0,
+            extent_x=100.0,
+            extent_y=100.0,
+        ),
+        depth=40.0,
+        duration=60.0,
+        dt=0.005,
+    )
+    srf_config = realisations.SRFConfig(
+        genslip_dt=1.0,
+        genslip_seed=1,
+        genslip_version="5.4.2",
+        resolution=0.1,
+        srfgen_seed=1,
+    )
+
+    realisation_ffp = tmp_path / "realisation.json"
+    domain_parameters.write_to_realisation(realisation_ffp)
+    srf_config.write_to_realisation(realisation_ffp)
+    with open(realisation_ffp, "r") as realisation_handle:
+        assert json.load(realisation_handle) == {
+            "domain": {
+                "resolution": 0.1,
+                "domain": [
+                    {"latitude": -43.524793866326725, "longitude": 171.76204128885567},
+                    {"latitude": -44.16756820707226, "longitude": 172.63312824122775},
+                    {"latitude": -43.53034935969409, "longitude": 173.51210368762364},
+                    {"latitude": -42.894200350955856, "longitude": 172.64076673694242},
+                ],
+                "depth": 40.0,
+                "duration": 60.0,
+                "dt": 0.005,
+            },
+            "srf": {
+                "genslip_dt": 1.0,
+                "genslip_seed": 1,
+                "resolution": 0.1,
+                "genslip_version": "5.4.2",
+                "srfgen_seed": 1,
+            },
+        }
+
+    assert realisations.SRFConfig.read_from_realisation(realisation_ffp) == srf_config
```

Comment:
Are these 2 asserts doing the same thing?

Reply:
not quite, one asserts that the json is what we expect it to be, the other asserts that the deserialisation logic and schema validation works correctly.